File: mobile/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String description;
  final double price;
  final String category;
  final String brand;
  final int stock;
  final String? imageUrl;
  final double? rating;
  final int ratingCount;
  final Map<String, String> specifications;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.category,
    required this.brand,
    required this.stock,
    this.imageUrl,
    this.rating,
    this.ratingCount = 0,
    this.specifications = const {},
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: (json['price'] as num).toDouble(),
      category: json['category'],
      brand: json['brand'],
      stock: json['stock'],
      imageUrl: json['imageUrl'],
      rating: (json['rating'] as num?)?.toDouble(),
      ratingCount: (json['ratingCount'] as num?)?.toInt() ?? 0,
      specifications: (json['specifications'] as Map<String, dynamic>? ?? {})
          .map((k, v) => MapEntry(k, '$v')),
    );
  }
}

File: backend/ShoppingAssistant.Api/Models/Product.cs
namespace ShoppingAssistant.Api.Models;

public class Product
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    /// <summary>Türk Lirası cinsinden satış fiyatı.</summary>
    public decimal Price { get; set; }

    public string Category { get; set; } = string.Empty;

    public string Brand { get; set; } = string.Empty;

    public int Stock { get; set; }

    public string? ImageUrl { get; set; }

    public double? Rating { get; set; }

    public int RatingCount { get; set; }

    /// <summary>Teknik özellikler, "Anahtar: değer; Anahtar: değer" biçiminde.</summary>
    public string? Specs { get; set; }

    /// <summary>Verinin geldiği kaynak: "seed", "dummyjson", "bestbuy" ...</summary>
    public string Source { get; set; } = "seed";

    /// <summary>Kaynaktaki ürün kimliği. (Source, ExternalId) tekildir.</summary>
    public string? ExternalId { get; set; }

    /// <summary>Ürünün kaynak sitedeki sayfası.</summary>
    public string? SourceUrl { get; set; }
}
